### DISCUSSION TITLE
Should I use getInitialProps or getServerSideProps

### DISCUSSION THREAD

**Discussion by ghost** [vercel/next.js#38994] [Help] [answered]
Hello there,
I'm making a Next.js (frontend) and Node.js (backend API) Project in which I have to fetch blog data. I'm trying to implement pagination with pages and filters with URL query parameters (?page=1&sort=likes) but I can't decide wether to use getInitialProps or getServerSideProps.

- If I would use getInitialProps I would save the request to Next.js server at continous requests for example to filter but also it makes the bundle size bigger. Also Better UX with a Loading Animation
- If I would use getServerSideProps it would make the bundle size smaller but introduces that extra request at every  page request.

What would you choose?

**Comment by valcosmos** [comment, marked as answer]
Recently, I rewrote my blog frontend project with `Next.js`, I use `getStaticProps` in post list and `getServerSideProps ` in post detail page.

You can actually try both and choose the one u think is perfect.  😊

**Reply by ghost** [reply]
Thanks, but getStaticProps doesn't have access to query params like ?test=123&abc=123, does it?

**Reply by valcosmos** [reply]
I have no idea, I am sorry, I haven't tried, maybe u can try it.

Anyway,  the documentation seems to recommend using `getStaticProps` and `getServerSideProps`, this seems to give better performance,  because getInitialProps will disable Automatic Static Optimization.
